### ISSUE TITLE
Creating a provider of ABC

### ISSUE THREAD

**Issue by awaizman1** [ets-labs/python-dependency-injector#266]
Hi,

I'm trying to create a Dependency provider for some python interface I have in the form of abstract-base-class (ABC).
Trying creating the provider raises an error.
What is the correct way for achieving this?
```python
# file testc.py

from abc import ABC

from dependency_injector import providers

class Base(ABC):
    pass

class Impl(Base):
    pass

p = providers.Dependency(Base)  # this line raises
p.provided_by(providers.Object(Impl()))
```
I get the following error:
```
Traceback (most recent call last):
  File "testc.py", line 13, in <module>
    p = providers.Dependency(Base)
TypeError: Argument 'instance_of' has incorrect type (expected type, got ABCMeta)
```

Thanks

**Comment by rmk135** [comment]
Hi @awaizman1 ,

That's a bug. It's easy to fix. Expect a fixed version on a couple of hours.

Roman

**Comment by awaizman1** [comment]
Thanks

appreciate it

**Comment by rmk135** [comment]
Fixed in `3.23.1`. It's already on the PyPI. You're welcome to upgrade.

Thanks for raising the issue.

Roman
